core/src/main/kotlin/org/evomaster/core/search/gene/StringGene.kt

```diff
@@ -32,13 +33,18 @@ class StringGene(
     }
 
     override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?, targetFormat: OutputFormat?): String {
-        when {
-            (targetFormat == null) -> return "\"$value\""
-             targetFormat.isKotlin() -> return "\"$value\""
-                                                            .replace("\\", "\\\\")
-                                                            .replace("$", "\\$")
-            else -> return "\"$value\""
-                    .replace("\\", "\\\\")
+        val rawValue = getValueAsRawString()
+        if (mode != null && mode.equals("xml")) {
+            return StringEscapeUtils.escapeXml(rawValue)
+        } else {
+            when {
+                (targetFormat == null) -> return "\"$rawValue\""
```

Comment:
add a  `// TODO: refactor`
at later stage, we will need a general way to handle with this issue

Reply:
ok